tockloader-lib/src/lib.rs

```diff
@@ -0,0 +1,149 @@
+// Licensed under the Apache License, Version 2.0 or the MIT License.
+// SPDX-License-Identifier: Apache-2.0 OR MIT
+// Copyright OXIDOS AUTOMOTIVE 2024.
+
+pub mod attributes;
+mod board_settings;
+mod errors;
+pub mod probe_session;
+pub mod tabs;
+
+use attributes::app_attributes::AppAttributes;
+use attributes::get_app_attributes::{get_apps_data, get_start_address};
+use attributes::get_board_attributes::{get_appaddr, get_board_attributes, get_bootloader_version};
+use attributes::get_kernel_attributes::{get_kernel_attributes, get_kernel_version};
+use attributes::hardware_attributes::HardwareAttributes;
+
+use tabs::tab::Tab;
+
+use errors::TockloaderError;
+use probe_rs::probe::DebugProbeInfo;
+use probe_session::ProbeSession;
+
+pub async fn list_probe(
+    choice: DebugProbeInfo,
+    board: &str,
+    chip: &str,
+    core_index: &usize,
+) -> Vec<AppAttributes> {
+    let mut probe_session = ProbeSession::new(choice, board, chip);
+    let mut core = probe_session.get_core(*core_index);
+
+    get_apps_data(&mut core)
+}
+
+pub async fn info_probe(
+    choice: DebugProbeInfo,
+    board: &str,
+    chip: &str,
+    core_index: &usize,
+) -> (HardwareAttributes, Vec<AppAttributes>) {
```

Comment:
I would avoid returning pairs. This isn't that bad since it's clear what they are. Please see coment for  the `hardware_attributes.rs` file.